network-api/networkapi/minisites/models.py

```diff
@@ -0,0 +1,360 @@
+from django.db import models
+
+from wagtail.core import blocks
+from wagtail.core.models import Page
+from wagtail.core.fields import StreamField, RichTextField
+from wagtail.admin.edit_handlers import FieldPanel, StreamFieldPanel
+from wagtail.snippets.edit_handlers import SnippetChooserPanel
+from wagtail.images.blocks import ImageChooserBlock
+from wagtail.snippets.models import register_snippet
+
+from adminsortable.models import SortableMixin
+
+
+class LinkButtonValue(blocks.StructValue):
+    # and https://stackoverflow.com/questions/49374083
+    # see http://docs.wagtail.io/en/v2.0/topics/streamfield.html#custom-value-class-for-structblock
+
+    @property
+    def css(self):
+        # Note that StructValue is a dict-like object, so `styling` and `outline`
+        # need to be accessed as dictionary keys
+        btn_class = self['styling']
+        if self['outline'] is True:
+            btn_class = btn_class.replace('btn-', 'btn-outline-')
+        return btn_class
+
+
+class LinkButtonBlock(blocks.StructBlock):
+    label = blocks.CharBlock()
+
+    # We use a char block because UrlBlock does not
```

Comment:
Is UrlBlock the component that results in this UI? 
<img width="1063" alt="image" src="https://user-images.githubusercontent.com/1682681/37778534-8c118bfe-2dc0-11e8-8dc7-6bbbbe2e5cd8.png">

Reply:
No, it's just a model property block for "this string counts as a URL" which is rendered on pages using an `<a...>` element. It does not have anything to do with admin functionality.